agent/consul/leader_connect_ca.go

```diff
@@ -970,6 +977,7 @@ func (c *CAManager) getIntermediateCAPrimary(provider ca.Provider, newActiveRoot
 	if err != nil {
 		return fmt.Errorf("error generating new intermediate cert: %v", err)
 	}
+	intermediatePEM = ca.AddSingleNewline(intermediatePEM)
```

Comment:
minor: I think by moving this immediate before the line which appends it, it should make it more clear why we are appending the newline.